src/coreclr/src/inc/corjit.h

```diff
@@ -251,6 +251,14 @@ class ICorJitInfo : public ICorDynamicInfo
         UINT32 ExecutionCount;
     };
 
+    struct ClassProfile
+    {
+        enum { SIZE = 8, SAMPLE_INTERVAL = 32 };
+        UINT32 ILOffset;
```

Comment:
Sure, will do. It's currently done so that we can allocate class profiles and block counts as one big chunk without telling the runtime how many of either exist. The high bit tells us this is a class profile, and the next bit which kind.

This magic will go away once we get more of the runtime side implemented and can specify a more flexible "schema" for the profile data.